content/memory_model.tex

```diff
@@ -54,16 +54,64 @@
 
 \subsection{Atomicity Guarantees}\label{subsec:amo_guarantees}
 
-\openshmem contains a number of routines that operate on symmetric data
-atomically (Section \ref{sec:amo}).  These routines guarantee that accesses by
-\openshmem's atomic operations with the same datatype will be exclusive, but do not guarantee
-exclusivity in combination with other routines, either inside \openshmem or
-outside.
+\openshmem contains a number of routines that perform atomic operations on
+symmetric data objects, which are defined in Section \ref{sec:amo}.  In
+addition, \openshmem defines routines that are atomic compatible, e.g. the
+point-to-point synchronization routines defined in
+Section~\ref{subsec:p2p_intro}.  The atomic and atomic compatible routines
+guarantee that concurrent accesses by any of these routines to the same
+location and using the same datatype (specified in Tables~\ref{stdamotypes} and
+\ref{extamotypes}) will be exclusive.  In contrast to atomic operations, atomic
```

Comment:
`short` and `unsigned short` is supported by `wait_until` (not supported in amotypes); `float` and `double` are the other non-overlapping types.

Reply:
This text is intended to specify semantics for atomics, so those types shouldn't really come into play.  But, I see your point that as it's written, it also specifies semantics for overlapping wait operations without atomics.  How about splitting these cases up:
> The atomic routines guarantee that concurrent accesses by any of these routines to the same location and using the same datatype (specified in Tables~\ref{stdamotypes} and \ref{extamotypes}) will be exclusive.  Similarly, concurrent accesses by atomic and atomic compatible routines to the same location and using the same datatype will also be exclusive.

Reply:
Follow-up question : will there be a usecase for `short` and `unsigned short` if the amo APIs end up being the only way to signal `wait_until`?

Reply:
IIRC, InfiniBand doesn't have hardware support for 8- and 16-bit atomics.  That's one argument in favor supporting scalar put with the wait operations..